source/dofs/dof_tools_constraints.cc

```diff
@@ -533,6 +533,14 @@ namespace DoFTools
                   for (unsigned int i = 0; i < n_primary_dofs; ++i)
                     abs_sum += std::abs(face_constraints(row, i));
 
+                  // if all entries of the row are zero, the corresponding dof
+                  // is not supposed to be constrained to any other dof. we can
+                  // thus skip this row
+                  // this happens for example whenever FE_System elements with
+                  // FE_Nothing elements are constrained
+                  if (abs_sum == 0)
+                    continue;
```

Comment:
As we discussed today, this isn't correct. It would trigger for the case of `FE_Q(1)` next to an `FE_Nothing` regardless of whether the latter dominates or not, I believe. This would probably be easy to verify with a little testcase, but I'm pretty sure this is not correct.